roles/os_hardening/tasks/cron.yml

```diff
@@ -0,0 +1,84 @@
+---
+# Granting write access to this directory for non-privileged users could provide
+# them the means for gaining unauthorized elevated privileges.
+# Granting read access to this directory could give an unprivileged user insight
+# in how to gain elevated privileges or circumvent auditing controls.
+# CIS 5.1.2 - CIS 5.1.7
+#
+- name: Check if /etc/crontab exists
+  stat:
+    path: /etc/crontab
+  register: osh_crontab
+
+- name: Ensure permissions on /etc/crontab are configured
+  ansible.builtin.file:
+    path: /etc/crontab
+    owner: root
+    group: root
+    mode: '0600'
+  when: osh_crontab.stat.exists
+
+- name: Check if /etc/cron.hourly exists
+  stat:
+    path: /etc/cron.hourly
+  register: osh_cron_hourly
+
+- name: Ensure permissions on /etc/cron.hourly are configured
```

Comment:
these should also all have `state: directory`
maybe also put all the directories in one loop, this makes it more compact.

Reply:
you can create a list of all existing cron directories by using this task:

```yaml
    - find:
        paths:
          - /etc
        patterns:
          - cron.hourly
          - cron.daily
          - cron.weekly
          - cron.monthly
          - cron.d
        file_type: directory
      register: cron_directories
```

Reply:
It modifies the permissions as expected without the `state: directory`.
Having the state there, the only difference would be if there was a file named /etc/cron.hourly there for some reason. My guess is that Ansible would error in that case if the `state: directory` was set and not error if not set.

Doing a loop gives nicer looking code.
Not having a loop gives nicer output.
This is my personal opinion and I prioritize the output.
If there is a loop and permissions for /etc/cron.weekly and /etc/cron.daily are changed there is only one change according to Ansible, since only one task changed.

Having one task for one file gives more verbose output where it's more obvious what changed.

This is a matter of taste. You decide, I obey ;-)

Reply:
You are right, both solutions are adequate. I rather like using less utility tasks and status variables. So could you please put this into a single loop?

I also think, that having less code duplication is better, since it avoids copy&paste errors. A bit duplication might be ok, but here we are doing 5 times exaclty the same.